sdk/identity/identity/src/credentials/cliCredential.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { logger } from '../util/logging';
+import { CliCredentialClient } from '../client/CliCredentialClient';
+import { TokenCredentialOptions } from "../client/identityClient";
+
+/**
+ * Provides the user access token and expire time
+ * with azure cli command "az account get-access-token" in powershell.  
+ */
+export class CliCredential implements TokenCredential {
+  private client: CliCredentialClient;
+  /**
+   * Creates an instance of the CliCredential class.
+   *
+   * @param options Options for configuring the client which makes the authentication request.
+   */
+  constructor(cliCredentialClient: CliCredentialClient = new CliCredentialClient()) {
```

Comment:
As @bterlson points out, this approach will leak the private type into the public interface. Instead, we might want to use an interface here.  You can see examples of this here: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/test/node/managedIdentityCredential.spec.ts#L103-L106

Additionally, rather than a default value, you can always assume that is CliCredentialClient, and only change it if an optional value is passed in.  This is similar to what you've written, but more clear.

One way to do this: 
```typescript  
interface CredentialClient {
    createProcess(commandString: string): Promise<...>;
}

class CliCredentialClient implements CredentialClient {
    public createProcess(commandString: String): Promise<...> {
        // ...
    }
}

interface CliCredentialOptions {
    cliCredentialClient?: CredentialClient;
}

class CliCredential {
    private client: CredentialClient;

    constructor(options?: CliCredentialOptions) {
        if (options && options.cliCredentialClient) {
            this.client = options.cliCredentialClient;
        } else {
            this.client = new CliCredentialClient();
        }
    }
}
```